File: Ilya-Dudnikov/term_2/block_4/task_2/bash/src/main/java/BashProject/shellcommand/commands/Exit.java
package BashProject.shellcommand.commands;

import BashProject.shellcommand.Command;

import java.nio.ByteBuffer;

public class Exit extends Command {
	@Override
	public ByteBuffer run(String ... args) {
		if (args.length == 0) {
			Runtime.getRuntime().exit(0);
		}
		if (args.length > 1) {
			return ByteBuffer.wrap(("Too many arguments" + System.lineSeparator()) .getBytes());
		}

		try {
			Runtime.getRuntime().exit(Integer.parseInt(args[0]));
		} catch (NumberFormatException e) {
			return ByteBuffer.wrap(("Argument must be an integer" + System.lineSeparator()).getBytes());
		}
		return ByteBuffer.wrap(new byte[0]);
	}
}

File: Semester 2/Block 4/Task 8/BashLib/Commands/Exit.cs
using BashLib.IO;

namespace BashLib.Commands
{
	public class Exit : ICommand
	{
		private IExiter exiter;
		
		public Exit(IExiter exiter)
		{
			this.exiter = exiter;
		}
		
		public string Run(string[] arguments)
		{
			exiter.Exit();
			return string.Empty;
		}
	}
}